src/main/java/htsjdk/samtools/SAMUtils.java

```diff
@@ -1096,4 +1097,79 @@ public static SAMRecord clipOverlappingAlignedBases(final SAMRecord record, fina
         return value >= 0 && value <= BinaryCodec.MAX_UINT;
     }
+    
+    /**
+     * Extract a List of 'other canonical alignments' from a SAM record. Those alignments are stored as a string in the 'SA' tag as defined
+     * in the SAM specification.
+     * Each record in the List is a non-paired read.
+     * The name, sequence and qualities are copied from the original record.
+     * The SAM flag is set to <code>SUPPLEMENTARY_ALIGNMENT (+ READ_REVERSE_STRAND )</code>
+     * @param record must be non null and must have a non-null associated header.
+     * @return a list of 'other canonical alignments' SAMRecords. The list is empty if the 'SA' attribute is missing.
+     */
+    public static List<SAMRecord> getOtherCanonicalAlignments(final SAMRecord record) {
+        final Pattern semiColonPattern = Pattern.compile("[;]");
+        final Pattern commaPattern = Pattern.compile("[,]");
+        if( record == null ) throw new IllegalArgumentException("record is null");
+        if( record.getHeader() == null ) throw new IllegalArgumentException("record.getHeader() is null");
+        /* extract value of SA tag */
+        final Object saValue = record.getAttribute( SAMTagUtil.getSingleton().SA );
+        if( saValue == null ) return Collections.emptyList();
+        if( ! (saValue instanceof String) ) throw new SAMException(
+                "Expected a String for attribute 'SA' but got " + saValue.getClass() );
+        
+        final SAMRecordFactory samReaderFactory = new DefaultSAMRecordFactory();
+        
+        /* the spec says: "Other canonical alignments in a chimeric alignment, formatted as a 
+         * semicolon-delimited list: (rname,pos,strand,CIGAR,mapQ,NM;)+. 
+         * Each element in the list represents a part of the chimeric alignment.
+         * Conventionally, at a supplementary line, the  1rst element points to the primary line.
+         */
+        
+        /* break string using semicolon */
+        final String semiColonStrs[] = semiColonPattern.split((String)saValue);
+        
+        /* the result list */
+        final List<SAMRecord> alignments = new ArrayList<>( semiColonStrs.length );
+        
+        for( final String semiColonStr : semiColonStrs ) {
+            /* ignore empty string */
+            if( semiColonStr.isEmpty() ) continue;
+            
+            /* break string using comma */
+            final String commaStrs[] = commaPattern.split(semiColonStr);
+            if( commaStrs.length != 6 )  throw new SAMException("Bad 'SA' attribute in " + semiColonStr);
+            
+            /* create the new record */
+            final SAMRecord otherRec = samReaderFactory.createSAMRecord( record.getHeader() );
+            
+            /* copy fields from the original record */
+            otherRec.setReadName( record.getReadName() );
+            otherRec.setReadBases( record.getReadBases() );
+            otherRec.setBaseQualities( record.getBaseQualities() );
```

Comment:
Perhaps you might also set the supplementary alignment flag in the new read? It should be predictable according to convention